components/Camera/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { Camera } from 'expo-camera'
+import { CameraImage } from 'types'
+import { AntDesign, FontAwesome, Ionicons, Feather } from '@expo/vector-icons'
+
+interface Props {
+  image: CameraImage | null
+  setImage: (newImage: CameraImage | null) => void
+  closeCamera: () => void
+}
+
+const ICON_SIZE = 40
+
+export default function CameraScreen({ image, setImage, closeCamera }: Props) {
+  const [hasPermission, setHasPermission] = useState<null | boolean>(null)
+  const appState = useRef(AppState.currentState)
+  const [appStateVisible, setAppStateVisible] = useState(appState.current === 'active')
+  const [type, setType] = useState(Camera.Constants.Type.back)
+  const [available, setAvailable] = useState<boolean>(true)
+  const cameraRef = useRef(null)
+
+  useEffect(() => {
+    const requestPermissions = async () => {
+      const { status } = await Camera.requestCameraPermissionsAsync()
+      setHasPermission(status === 'granted')
+    }
+    const cameraIsAvailable = async () => {
+      let available = await Camera.isAvailableAsync()
+      setAvailable(available)
+    }
+    if (Platform.OS === 'web') {
+      cameraIsAvailable()
+    }
+    requestPermissions()
+  }, [])
+
+  useEffect(() => {
+    AppState.addEventListener('change', handleNextAppState)
+
+    return () => {
+      AppState.removeEventListener('change', handleNextAppState)
+    }
+  }, [])
+
+  const handleSwitchCameraType = () => {
+    setType(type === Camera.Constants.Type.back ? Camera.Constants.Type.front : Camera.Constants.Type.back)
+  }
+
+  const handleNextAppState = (nextAppState: AppStateStatus) => {
+    appState.current = nextAppState
+    setAppStateVisible(appState.current === 'active')
+  }
+
+  if (hasPermission === null) {
+    return <View />
+  }
+
+  if (hasPermission === false) {
+    return <Text>No access to camera</Text>
+  }
+
+  if (!available) {
+    return <Text>Device has no camera</Text>
+  }
+
+  return (
+    <View style={styles.container}>
+      {image ? (
+        <ImageBackground source={{ uri: image.uri }} style={styles.takenPhoto}>
+          <View style={styles.buttonContainer}>
+            <FontAwesome name="rotate-left" size={ICON_SIZE} color="white" onPress={() => setImage(null)} />
+            <Feather name="check-circle" size={ICON_SIZE} color="white" onPress={closeCamera} />
+          </View>
+        </ImageBackground>
+      ) : (
+        appStateVisible && (
+          <Camera style={styles.camera} type={type} ratio="8:10" ref={cameraRef}>
+            <View style={styles.cameraContainer}>
+              <AntDesign name="close" size={ICON_SIZE} color="white" onPress={closeCamera} style={styles.closeButton} />
+              <View style={styles.buttonContainer}>
+                <AntDesign
+                  onPress={async () => {
+                    if (cameraRef.current) {
+                      // @ts-ignore
```

Comment:
Nit: You can probably remove this `@ts-ignore` by setting the `useRef` type to `useRef<Camera>(null)`

Reply:
thanks, i'll look into the preview

Reply:
I think it has to do with the aspect ratio, I chose 4:5 as that's a typical portrait 8x10 standard, so if the image isn't cropped in that ratio is messes up :(

Reply:
I think so too and looking into it the fixes look pretty awkward :( I think we can just merge it in and fix it after the demo if that suits better.

Reply:
> I think it has to do with the aspect ratio, I chose 4:5 as that's a typical portrait 8x10 standard, so if the image isn't cropped in that ratio is messes up :(

I'm not sure if this is what we want, but apparently, we can get the applications' height and width to calculate the aspect ratio dynamically at runtime. 
[here](https://docs.expo.dev/versions/latest/react-native/dimensions/)
[there](https://www.youtube.com/watch?v=fI6xyzNm0TY)

Reply:
I got a fix for it, essentially all you have to do is make sure whatever <Container> the camera is in, it conforms to whatever aspect ratio is set for the camera

Reply:
Awesome! Looks like it works well, thanks for the quick fix!

![image](https://user-images.githubusercontent.com/29843837/154261419-e25ee437-aa2d-4de2-8dd5-2a87492825d0.png)